### PR TITLE
Refactor WebSub resource detail map generation

### PR DESCRIPTION
**Pull request by MaryamZi**
## Purpose
This PR refactors WebSub to
- improve resource detail map generation for custom webhooks
- fix warnings

### CODE REVIEW

File: stdlib/websub/src/main/java/org/ballerinalang/net/websub/nativeimpl/InitWebSubSubscriberServiceEndpoint.java [hunk 1]
```diff
@@ -110,14 +120,93 @@ public void execute(Context context) {
...
+        });
+    }
+
+    private static void populateResourceDetailsByHeaderAndPayload(BMap<String, BMap<String, BMap<String, BValue>>>
+                                                                          headerAndPayloadKeyResourceMap,
+                                                                  HashMap<String, String[]> resourceDetails) {
+        headerAndPayloadKeyResourceMap.getMap().values().forEach(mapByHeader -> {
+            mapByHeader.getMap().values().forEach(mapByKey -> {
+                mapByKey.getMap().values().forEach(value -> {
+                    BRefValueArray resourceDetailTuple = (BRefValueArray) value;
```
Comment (chamil321) [inline stdlib/websub/src/main/java/org/ballerinalang/net/websub/nativeimpl/InitWebSubSubscriberServiceEndpoint.java]: Shall we move this redundant code in to another method as follows?

```ballerina
private static void populateResourceDetails(HashMap<String, String[]> resourceDetails, BRefValueArray resourceDetailTuple) {
        String resourceName = resourceDetailTuple.getBValue(0).stringValue();
        BStructureType paramDetails =
                (BStructureType) ((BTypeDescValue) (resourceDetailTuple).getBValue(1)).value();
        resourceDetails.put(resourceName,
                new String[]{paramDetails.getPackagePath(), paramDetails.getName()});
    }
```

File: stdlib/websub/src/main/java/org/ballerinalang/net/websub/nativeimpl/InitWebSubSubscriberServiceEndpoint.java [hunk 2]
```diff
@@ -110,14 +120,84 @@ public void execute(Context context) {
...
+        });
+    }
+
+    private static void populateResourceDetails(HashMap<String, String[]> resourceDetails,
+                                                BRefValueArray resourceDetailTuple) {
+        String resourceName = resourceDetailTuple.getBValue(0).stringValue();
+        BStructureType paramDetails =
+                (BStructureType) ((BTypeDescValue) (resourceDetailTuple).getBValue(1)).value();
+        resourceDetails.put(resourceName,
+                            new String[]{paramDetails.getPackagePath(), paramDetails.getName()});
```
Comment (anupama-pathirage) [inline stdlib/websub/src/main/java/org/ballerinalang/net/websub/nativeimpl/InitWebSubSubscriberServiceEndpoint.java]: Reformat the code. Has unnecessary double lines

File: stdlib/websub/src/main/java/org/ballerinalang/net/websub/nativeimpl/InitWebSubSubscriberServiceEndpoint.java (hunk 1 above)
Comment (MaryamZi) [reply stdlib/websub/src/main/java/org/ballerinalang/net/websub/nativeimpl/InitWebSubSubscriberServiceEndpoint.java]: Fixed with e631a56, thanks.

File: stdlib/websub/src/main/java/org/ballerinalang/net/websub/nativeimpl/InitWebSubSubscriberServiceEndpoint.java (hunk 2 above)
Comment (MaryamZi) [reply stdlib/websub/src/main/java/org/ballerinalang/net/websub/nativeimpl/InitWebSubSubscriberServiceEndpoint.java]: Fixed with 8da454e, thanks.

**Pull request merged**